### PR TITLE
Fix to log nz_azconnector utility error on terminal 

### PR DESCRIPTION
**Pull request by shabbir10july**
JIRA: https://github.com/IBM/netezza-utils/issues/7

Problem: Currently the utility writes all errors only to log file. On terminal, we do not know if it was successful or failed with error.

Solution: 
1. Added fmt.print to print error/message on terminal.
2. Fixed minor issue related to blob not found

Testing: Tried upload/download database backup from filesystem to cloud and vice versa. Tried with some invalid combination to produce errors. Logs present in JIRA.

### CODE REVIEW

**Review by aniket-s-kulkarni — CHANGES_REQUESTED** [review]
@shabbir10july pl change line https://github.com/IBM/netezza-utils/pull/8/files#diff-ca87319174ba02d1c7d8d4fe21b560f60d44a85677d87a6a746574b90c707a5aR329 to say

```go
w = io.Multiwriter(os.Stdout, filehandle)
log.SetOutput(w)
```

And remove all other changes.

File: bnr-utils/nz_azConnector/nz_azConnector.go [hunk 1]
```diff
@@ -102,6 +102,7 @@ func parseArgs(conn *Conn, backupinfo *BackupInfo, othargs *OtherArgs) {
 
 func handleErrors(err error) {
     if err != nil {
+        fmt.Println(err)
```
Comment (brajeshkpandey) [inline bnr-utils/nz_azConnector/nz_azConnector.go]: we should prepend error also if not already done .. 

> fmt.Printf("error: %s", err)

File: bnr-utils/nz_azConnector/nz_azConnector.go [hunk 2]
```diff
@@ -239,7 +240,8 @@ func (cn *Conn) downloadBkp(outdir string, uniqueid string, blobpath string, str
                 }
                 if r.err != nil {
                     // stopping right here so that we
-                    // don't keep on uploading when one has failed
+                    // don't keep on downloading when one has failed
+                    fmt.Printf("%s: %v", r.blobname, r.err)
```
Comment (brajeshkpandey) [inline bnr-utils/nz_azConnector/nz_azConnector.go]: same here

File: bnr-utils/nz_azConnector/nz_azConnector.go [hunk 3]
```diff
@@ -404,6 +406,7 @@ func main() {
             close(work)
             <- done
             handleErrors(err)
+            fmt.Println("Upload successful. Total files uploaded:", filesuploaded)
             log.Println("Upload successful. Total files uploaded:", filesuploaded)
         }
```
Comment (brajeshkpandey) [inline bnr-utils/nz_azConnector/nz_azConnector.go:427]: general, we should set logger as Aniket said to write at both the places.

File: bnr-utils/nz_azConnector/nz_azConnector.go (hunk 1 above)
Comment (shabbir10july) [reply bnr-utils/nz_azConnector/nz_azConnector.go]: Done

**Comment by shabbir10july** [comment]
> @shabbir10july pl change line https://github.com/IBM/netezza-utils/pull/8/files#diff-ca87319174ba02d1c7d8d4fe21b560f60d44a85677d87a6a746574b90c707a5aR329 to say
> 
> ```go
> w = io.Multiwriter(os.Stdout, filehandle)
> log.SetOutput(w)
> ```
> 
> And remove all other changes.

Done

File: bnr-utils/nz_azConnector/nz_azConnector.go (hunk 3 above)
Comment (shabbir10july) [reply bnr-utils/nz_azConnector/nz_azConnector.go:427]: Done

File: bnr-utils/nz_azConnector/nz_azConnector.go (hunk 2 above)
Comment (shabbir10july) [reply bnr-utils/nz_azConnector/nz_azConnector.go]: Done

File: bnr-utils/nz_azConnector/nz_azConnector.go [hunk 4]
```diff
@@ -349,7 +347,9 @@ func main() {
             log.Println("Uploading backup data to azure cloud from backup dir", bkpdir)
             backupdir := filepath.Join(bkpdir, "Netezza", backupinfo.npshost, backupinfo.dbname, backupinfo.backupsetID)
             _, err = os.Stat(backupdir)
-            handleErrors(err)
+            if err != nil {
+                handleErrors(fmt.Errorf("Unable to traverse existing backup on file system: %v",err))
```
Comment (aniket-s-kulkarni) [inline bnr-utils/nz_azConnector/nz_azConnector.go]: Please use this

```suggestion
                handleErrors(fmt.Errorf("Cannot access directory '%s': %v", backupdir, err))

```

File: bnr-utils/nz_azConnector/nz_azConnector.go [hunk 5]
```diff
@@ -384,7 +384,8 @@ func main() {
                         if r.err != nil {
                             // stopping right here so that we
                             // don't keep on uploading when one has failed
-                            log.Fatalf("%s: %v", r.job, r.err)
+                            log.Println("Error while uploading file. Ensure azure storage account name, azure key and container name are correct. If error persists contact IBM support team.")
```
Comment (aniket-s-kulkarni) [inline bnr-utils/nz_azConnector/nz_azConnector.go]: @shabbir10july lets make sure our uploadJob and downloadJob can be printed correctly.

```go
// at the top of the file

func (c Conn) String() string {
    return fmt.Sprintf("account:%s container:%s", c.azaccount, c.azcontainer)
}

func (j Job) String() string {
    return fmt.Sprintf("conn:[%s] backupDir:%s id:%s", j.conn, j.backupdir, j.uniqueid)
}

func (u uploadJob) String() string {
    return fmt.Sprintf("%s file:%s", u.job, u.absfilepath)
}

func (d downloadJob) String() string {
    return fmt.Sprintf("conn:[%s] blob:%s file:%s", d.conn, d.blobname, d.outfilepath)
}
```

With that change these two lines to

```go
log.Fatalf("Error uploading file. Ensure azure storage account name, azure key and container name are correct. If error persists contact IBM support team: %s`,
    *r.job)
```

File: bnr-utils/nz_azConnector/nz_azConnector.go [hunk 6]
```diff
@@ -403,7 +404,9 @@ func main() {
                 })
             close(work)
             <- done
-            handleErrors(err)
+            if err != nil {
+                handleErrors(fmt.Errorf("Unable to traverse existing backup on file system: %v",err))
```
Comment (aniket-s-kulkarni) [inline bnr-utils/nz_azConnector/nz_azConnector.go]: Pl use

```suggestion
                handleErrors(fmt.Errorf("Error reading directory: %s : %v", backupdir, err))

```

**Review by aniket-s-kulkarni — CHANGES_REQUESTED** [review]

File: bnr-utils/nz_azConnector/nz_azConnector.go (hunk 4 above)
Comment (shabbir10july) [reply bnr-utils/nz_azConnector/nz_azConnector.go]: Done

File: bnr-utils/nz_azConnector/nz_azConnector.go (hunk 6 above)
Comment (shabbir10july) [reply bnr-utils/nz_azConnector/nz_azConnector.go]: Done

File: bnr-utils/nz_azConnector/nz_azConnector.go (hunk 5 above)
Comment (shabbir10july) [reply bnr-utils/nz_azConnector/nz_azConnector.go]: okay done

**Review by aniket-s-kulkarni — APPROVED** [review]

**Review by abhishekjog — APPROVED** [review]

**Comment by shabbir10july** [comment]
@brajesh1984 Could you please review and approve the changes

**Review by aniket-s-kulkarni — APPROVED** [review]
lgtm

**Review by brajeshkpandey — APPROVED** [review]
LGTM .. thanks

**Pull request merged**